test/Web3Spec/Encoding/DataSpec.purs

```diff
@@ -0,0 +1,59 @@
+module Web3Spec.Encoding.DataSpec (dataMakerSpec) where
+
+-- There is no test here, this just needs to compile
+
+import Prelude
+import Data.Maybe (fromJust)
+import Network.Ethereum.Web3.Solidity (UIntN, Tuple2, uIntNFromBigNumber)
+import Network.Ethereum.Web3.Solidity.Size (type (:&), D2, D5, D6, DOne)
+import Network.Ethereum.Web3.Solidity.Sizes (s256)
+import Network.Ethereum.Web3.Types (Address, HexString, TransactionOptions, NoPay, Web3, mkHexString, mkAddress)
+import Data.Functor.Tagged (Tagged, tagged)
+import Data.Symbol (class IsSymbol, SProxy(..), reflectSymbol)
+import Network.Ethereum.Web3.Contract (sendTx)
+import Network.Ethereum.Core.Keccak256 (toSelector)
+import Network.Ethereum.Web3.Solidity.Generic (class RecordFieldsIso, class GenericABIEncode, genericFromRecordFields, genericABIEncode)
+import Data.Generic.Rep (class Generic, Constructor)
+import Type.Proxy (Proxy(..))
+import Partial.Unsafe (unsafePartial)
+
+import Network.Ethereum.Web3.Solidity.AbiEncoding (toDataBuilder)
+
+import Test.Spec (Spec, describe, it)
+import Test.Spec.Assertions (shouldEqual)
+
+
+dataMakerSpec :: forall r . Spec r Unit
+dataMakerSpec =
+    describe "data maker" do
+
+      it "can make the approval data" do
+        let addr = unsafePartial fromJust $ (mkAddress =<< mkHexString "78534a937a855e15be172de35f2211626f92f8ec")
+            val = unsafePartial fromJust $ uIntNFromBigNumber s256 one
+            approvalD = dataMaker (Proxy :: Proxy ApproveFn) {_spender: addr, _value: val}
+            sel = toSelector "approve(address,uint256)"
+            fullDat = sel <> toDataBuilder addr <> toDataBuilder val
+        approvalD `shouldEqual` fullDat
+
```

Comment:
We can extract the selector creation into a function:

``` purs
getSelector :: forall s a. IsSymbol s => Proxy (Tagged (SProxy s) a) -> HexString
-- it could be even more general
-- x :: forall f s a. IsSymbol s => Proxy (f (SProxy s) a) -> HexString
getSelector _ = toSelector $ reflectSymbol $ SProxy :: SProxy s

selector :: HexString
selector = getSelector $ Proxy :: Proxy ApproveFn
```

We could probably add this `getSelector` to some module and use it here, or we can add it later.